### PR TITLE
Reimplement clone interface of expression

### PR DESCRIPTION
**Pull request by yixinglu**
Replace decode/encode with construction

Disable variable expression decode function.

### CODE REVIEW

File: src/common/expression/Expression.cpp [hunk 1]
```diff
@@ -401,13 +397,11 @@ std::unique_ptr<Expression> Expression::decode(Expression::Decoder& decoder) {
             return exp;
         }
         case Expression::Kind::kVar: {
-            exp = std::make_unique<VariableExpression>();
```
Comment (Shylock-Hg) [inline src/common/expression/Expression.cpp:404]: `InputProperty` And `VariableProperty` should disabled too.

**Review by dutor — DISMISSED** [review]
LGTM.

File: src/common/expression/Expression.cpp (hunk 1 above)
Comment (yixinglu) [reply src/common/expression/Expression.cpp:404]: fixed

**Review by Shylock-Hg — DISMISSED** [review]

**Comment by darionyaphet** [comment]
Sorry why `expression` should support clone ?

**Comment by Shylock-Hg** [comment]
> Sorry why `expression` should support clone ?

Used for rewrite some AST.

**Review by Shylock-Hg — APPROVED** [review]

**Review by dutor — APPROVED** [review]

**Pull request merged**